mautic/mautic
# Filter report data by tags

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR adds possibility to filter reports with contacts by tags.

Tag filter is available in these data sources:
* Emails Sent (+ added segment filter)
* Contacts
* Multi Touch Attributions
* First Touch Attributions
* Last Touch Attributions
* Contact Point Log
* Frequency Rules
* Segment Membership
* Do Not Contact (+ added segment filter)
* Contact's segments activity log
* UTM codes (+ added segment filter)




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add few contacts and assign tags
3. Create new Report in Reports page
4. Select Contacts (or other data source with tag filter) in "Data Source" drop down in Details tab
5. Click Add filter button to add new filter
6. Select Tags in "Column" select box and add required options
7. Save and close the report to see results




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11346"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
